tests/tests/src/lampo_cln_tests.rs

```diff
@@ -539,3 +540,373 @@ fn be_able_to_kesend_payments() {
+        if !channels.channels.first().unwrap().ready {
+            return Err(());
+        }
+        let address = cln.rpc().newaddr(None).unwrap();
+        fund_wallet(btc.clone(), &address.bech32.unwrap(), 1).unwrap();
+        crate::wait_cln_sync!(cln);
+        Err(())
+    });
+
+    let channels: response::Channels = lampo.call("channels", json::json!({})).unwrap();
+
+    let result: Result<response::CloseChannel, _> = lampo.call(
+        "close",
+        request::CloseChannel {
+            node_id: info_cln.id.to_string(),
+            channel_id: None,
+        },
+    );
+    assert!(result.is_ok(), "{:?}", result);
+    assert_eq!(
+        result.as_ref().unwrap().channel_id,
+        channels.channels.first().unwrap().channel_id.to_string()
+    );
+    assert_eq!(
+        result.as_ref().unwrap().peer_id,
+        channels.channels.first().unwrap().peer_id.to_string()
+    );
+    async_run!(cln.stop()).unwrap();
+}
+
+#[test]
+fn test_close_channel_without_opening_a_channel_success() {
+    init();
+    let mut cln = async_run!(cln::Node::with_params(
+        "--developer --dev-bitcoind-poll=1 --dev-fast-gossip --dev-allow-localhost",
+        "regtest"
+    ))
+    .unwrap();
+    let btc = cln.btc();
+    let lampo_manager = LampoTesting::new(btc.clone()).unwrap();
+    let lampo = lampo_manager.lampod();
+    let _info: response::GetInfo = lampo.call("getinfo", json::json!({})).unwrap();
+    let info_cln = cln.rpc().getinfo().unwrap();
+    let events = lampo.events();
+    lampo_manager.fund_wallet(101).unwrap();
+    wait!(|| {
+        let Ok(Event::OnChain(OnChainEvent::NewBestBlock((_, height)))) =
+            events.recv_timeout(Duration::from_millis(100))
+        else {
+            return Err(());
+        };
+        if height.to_consensus_u32() == 101 {
+            return Ok(());
+        }
+        Err(())
+    });
+
+    // Closing an un opened channel
+    let result: Result<response::CloseChannel, _> = lampo.call(
+        "close",
+        request::CloseChannel {
+            node_id: info_cln.id.to_string(),
+            channel_id: None,
+        },
+    );
+    assert!(result.is_err(), "{:?}", result);
+    async_run!(cln.stop()).unwrap();
+}
+
+#[test]
+fn channel_id_tests() {
+    let node_id = "039c108cc6777e7d5066dfa33c611c32e6baa1c49de6d546b5b76686486d0360ac".to_string();
+
+    // This is a correct channel_hex of 32 bytes
+    let channel_hex =
+        Some("0a44677526ac8c607616bd91258d7e5df1d86fae9c32e23aa18703a650944c64".to_string());
+    let req = CloseChannel {
+        node_id: node_id.clone(),
+        channel_id: channel_hex,
+    };
+    let channel_bytes = [
+        10, 68, 103, 117, 38, 172, 140, 96, 118, 22, 189, 145, 37, 141, 126, 93, 241, 216, 111,
+        174, 156, 50, 226, 58, 161, 135, 3, 166, 80, 148, 76, 100,
+    ];
+    let channel_id_bytes = req.channel_id();
+    assert_eq!(channel_bytes, channel_id_bytes.unwrap().0);
+}
```

Comment:
This can be added inside lampo-common/src/model/close_channel.rs

inside the tests there are functional tests, and this is a unit test